Generator panics when `data` is not set at all

```
E1206 16:43:02.460508       1 runtime.go:66] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:72
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:65
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:58
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/hashmap.go:487
/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:157
/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:97
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:173
<autogenerated>:50
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:251
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:131
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:102
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:96
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:97
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:52
/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:102
/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:105
/usr/local/go/src/runtime/proc.go:185
/usr/local/go/src/runtime/asm_amd64.s:2197
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:58 +0x126
panic(0x131bd80, 0xc420482ac0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:58 +0x126
panic(0x131bd80, 0xc420482ac0)
	/usr/local/go/src/runtime/panic.go:489 +0x2cf
main.(*GeneratorController).SecretAdded(0xc4201799a0, 0x145b9c0, 0xc42051a480)
	/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:157 +0x512
main.(*GeneratorController).SecretAdded-fm(0x145b9c0, 0xc42051a480)
	/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:97 +0x3e
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(0xc4204829b0, 0xc4204829c0, 0x14e4a08, 0x145b9c0, 0xc42051a480)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:173 +0x49
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.(*ResourceEventHandlerFuncs).OnAdd(0xc4201799c0, 0x145b9c0, 0xc42051a480)
	<autogenerated>:50 +0x73
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.NewInformer.func1(0x1334a80, 0xc420281260, 0x1334a80, 0xc420281260)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:251 +0x24d
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc420084420, 0xc4203824b0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420 +0x218
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.(*Controller).processLoop(0xc4202e85b0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:131 +0x3c
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.(*Controller).(github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.processLoop)-fm()
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x2a
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc420571ea0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:96 +0x5e
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc42064fea0, 0x3b9aca00, 0x0, 0x129a901, 0xc4200a53e0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:97 +0xad
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc420571ea0, 0x3b9aca00, 0xc4200a53e0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/pkg/util/wait/wait.go:52 +0x4d
github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache.(*Controller).Run(0xc4202e85b0, 0xc4200a53e0)
	/go/src/github.com/mittwald/kubernetes-secret-generator/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x1c7
main.main()
	/go/src/github.com/mittwald/kubernetes-secret-generator/main.go:105 +0x490
```

--- a/main.go
+++ b/main.go
@@ -153,6 +153,10 @@ func (c *GeneratorController) SecretAdded(obj interface{}) {
 		delete(secretCopy.Annotations, SecretRegenerateAnnotation)
 	}
 
+	if secretCopy.Data == nil {
+		secretCopy.Data = make(map[string][]byte)
+	}
+
 	secretCopy.Annotations[SecretGeneratedAtAnnotation] = time.Now().String()
 	secretCopy.Data[val] = []byte(newPassword)
 

